src/MongoDB.Driver.Core/Core/Authentication/Rfc2898DeriveBytes.cs

```diff
@@ -0,0 +1,333 @@
+﻿/* Original work:
```

Comment:
From the CoreFX project.

Reply:
Changes:
 - Made class internal instead of public.
 - Defined strings that were originally defined in an external string resources class.
  - Added formatting function that was originally defined in above mentioned class.